src/i_dot_ai_utilities/logging/structured_logger.py

```
@@ -205,6 +211,9 @@ def _should_ship_logs(self, options: LoggerConfigOptions) -> bool:
 
         return selected_option
 
+    def _normalise_kwargs(self, **kwargs: Any) -> dict[str, str]:
+        return {k: json.dumps(v, ensure_ascii=False) if isinstance(v, (dict | list)) else v for k, v in kwargs.items()}
```

Can we use `isinstance(v, (dict, list))` in place of `isinstance(v, (dict | list))` for broader compatibility

---

There’s quite a lot going on in the return statement, and it might miss any silent failures during JSON serialisation. Could we add some exception handling around this to make it more robust?

`json.dumps(v, ensure_ascii=False) if isinstance(v, (dict | list)) else v`